html/modules/custom/reliefweb_api/reliefweb_api.module

```diff
@@ -230,7 +230,7 @@ function reliefweb_api_check_reindexing(EntityInterface $entity) {
   }
 
   foreach ($fields as $field) {
-    if ($entity->get($field)->getValue() != $original_entity->get($field)->getValue()) {
+    if (!$entity->get($field)->equals($original_entity->get($field))) {
```

Comment:
The direct comparison `$entity->get($field)->getValue() != $original_entity->get($field)->getValue()` was causing any source being saved even without changes to be queued for re-indexing because the homepage value array in the entity was missing an "attributes" property (because it's not used) that was present in the `original` entity.

(Basically the `$entity` fields contain only the necessary properties without the computed ones or the empty ones.)